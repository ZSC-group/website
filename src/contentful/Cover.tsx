import { documentToReactComponents } from "@contentful/rich-text-react-renderer"
import { css } from "@emotion/react"
import {
  flex,
  flexRow,
  fonts,
  WHEN_DESKTOP,
  WHEN_MOBILE,
  WHEN_TABLET,
  WHEN_TABLET_AND_UP,
  whiteText,
} from "src/estyles"
import { GridRow } from "src/layout/Grid2"
import { TABLET_BREAKPOINT } from "src/shared/Styles"
import { CoverContentType } from "src/utils/contentful"
import renderers from "src/contentful/nodes/enodes"
import Button, { SIZE } from "src/shared/Button.3"
import { useScreenSize } from "src/layout/ScreenSize"
import { BLOCKS } from "@contentful/rich-text-types"

const OPTIONS = {
  renderNode: {
    ...renderers,
    [BLOCKS.HEADING_1]: (_, children: string) => {
      return <h2 css={fonts.h1}>{children}</h2>
    },
  },
}

export default function Cover(props: CoverContentType) {
  const { isMobile } = useScreenSize()
  const size = isMobile
    ? props.imageMobile?.fields?.file?.details.image
    : props.imageDesktop?.fields?.file?.details.image
  return (
    <GridRow
      columns={2}
      darkMode={props.darkMode}
      wrapperCss={wrapperCss}
      css={props.illoFirst ? imageFirstRootCss : rootCss}
    >
      <div css={contentCss}>
        <h1
          css={css(
            props.superSize ? titleCss : fonts.h1,
            centerMobileCss,
            props.darkMode && whiteText
          )}
        >
          {props.title}
        </h1>
        <span css={props.darkMode ? subtitleDarkMode : centerMobileCss}>
          {documentToReactComponents(props.subTitle, OPTIONS)}
        </span>
        <div css={linkAreaCss}>
          {props.links?.map((link) => (
            <Button
              align={"center"}
              key={link.sys.id}
              size={isMobile ? SIZE.fullWidth : SIZE.normal}
              kind={link.fields.kind}
              text={link.fields.words}
              href={link.fields.href}
            />
          ))}
        </div>
      </div>
      <div css={illoCss}>
        <picture>
          <source
            media={`(max-width: ${TABLET_BREAKPOINT}px)`}
            srcSet={props.imageMobile?.fields.file.url}
          />
          <img
            width={size?.width}
            height={size?.height}
            css={css(

              props.illoFirst ? imageFirstCss : imageCss,

              props.verticalPosition === "flushBottomText" && flushBottomCss

            )}
            src={props.imageDesktop?.fields.file.url}
            alt={props.imageDesktop?.fields.description}
          />
        </picture>
      </div>
    </GridRow>
  )
}

const wrapperCss = css(flex, {
  alignItems: "center",
  justifyContent: "center",
  [WHEN_MOBILE]: {
    alignContent: "center",
    minHeight: "100vh",
  },
  [WHEN_TABLET]: {
    minHeight: "100vh",
    height: "fit-content",
  },
  [WHEN_DESKTOP]: {
    height: "100vh",
    minHeight: "fit-content",
    maxHeight: "80vw",
  },
})

const rootCss = css({
  gridTemplateAreas: `"content illo"`,
  overflow: "visible",
  [WHEN_MOBILE]: {
    alignContent: "center",
    flexDirection: "column-reverse",
  },
})

const imageFirstRootCss = css(rootCss, {
  gridTemplateAreas: `"illo content"`,
})

const titleCss = css(fonts.h1, {
  [WHEN_DESKTOP]: {
    fontSize: 68,
    lineHeight: "68px",
  },
  [WHEN_MOBILE]: {
    textAlign: "center",
    fontSize: 36,
    lineHeight: "38px",
  },
})

const centerMobileCss = css({
  [WHEN_MOBILE]: {
    textAlign: "center",
  },
})

const subtitleDarkMode = css(whiteText, centerMobileCss, {
  "h1, h2, h3, h4, p": whiteText,
})

const contentCss = css(flex, {
  justifySelf: "center",
  justifyContent: "center",
  flex: 1,
  gridArea: "content",
  [WHEN_TABLET_AND_UP]: {
    paddingTop: 56,
    paddingBottom: 48,
    minWidth: 320,
  },
  [WHEN_MOBILE]: {
    padding: 16,
    maxWidth: 450,
    alignSelf: "center",
  },
})

const linkAreaCss = css(flexRow, {
  [WHEN_MOBILE]: {
    flexDirection: "column",
    "& > div": {
      marginBottom: 24,
    },
  },
  [WHEN_TABLET_AND_UP]: {
    "& > div": {
      marginRight: 24,
      justifyContent: "center",
    },
  },
})

const illoCss = css({
  display: "flex",
  gridArea: "illo",
  position: "relative",
  [WHEN_MOBILE]: {
    width: "100vw",
    justifyContent: "center",
    alignItems: "center",
  },
})

const imageCss = css({
  overflow: "visible",
  [WHEN_MOBILE]: {
    overflow: "inherit",
  },

})

const flushBottomCss = css({
  [WHEN_TABLET_AND_UP]: {
    bottom: 0,
    position: "absolute",
  },
})

const imageFirstCss = css(imageCss, {
  right: 0,
})
