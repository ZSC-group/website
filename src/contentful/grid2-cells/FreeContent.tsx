import { css, CSSObject } from "@emotion/react"
import { documentToReactComponents } from "@contentful/rich-text-react-renderer"
import { flex, fonts, WHEN_MOBILE } from "src/estyles"
import renderNode from "src/contentful/nodes/enodes"
import { BLOCKS, INLINES, Block, Document } from "@contentful/rich-text-types"
import { BUTTON } from "src/contentful/nodes/embeds/BUTTON"
import { GALLARY } from "src/contentful/nodes/embeds/GALLARY"
import { ROW } from "../nodes/embeds/ROW"
import { Asset } from "contentful"

const EMBEDDABLE = {
  ...BUTTON,
  ...GALLARY,
  ...ROW,
}

function embedded(node: Block) {
  const contentType = node.data?.target?.sys?.contentType?.sys?.id
  const renderer = EMBEDDABLE[contentType]

  if (renderer) {
    return renderer(node.data.target)
  } else {
    console.info(contentType)
    return null
  }
}

const OPTIONS = {
  renderNode: {
    ...renderNode,
    [BLOCKS.HEADING_1]: (_, children: string) => {
      return <h2 css={h1ResponsiveCss}>{children}</h2>
    },
    [BLOCKS.EMBEDDED_ENTRY]: embedded,
    [INLINES.EMBEDDED_ENTRY]: embedded,
  },
}

const h1ResponsiveCss = css(fonts.h1, { [WHEN_MOBILE]: fonts.h1Mobile })
interface Props {
  colSpan: number
  body: Document
  cssStyle?: CSSObject
  darkMode: boolean
  listStyleImage?: Asset
}

export function FreeContent({ colSpan, body, cssStyle, darkMode, listStyleImage }: Props) {
  const customBullets = listStyleImage
    ? {
        ul: {
          listStyleImage: `url(${listStyleImage.fields.file.url})`,
        },
      }
    : null

  return (
    <div css={css(rootCss, { gridColumn: `span ${colSpan}` })}>
      <div css={css(flex, darkMode && darkModeText, cssStyle, customBullets)}>
        {documentToReactComponents(body, OPTIONS)}
      </div>
    </div>
  )
}

const rootCss = css(flex, {
  maxWidth: "calc(100vw - 16px)",
  a: {
    marginTop: 16,
  },
  "h2:first-of-type": {
    marginTop: 8,
  },
})

const darkModeText = css({ "h1, h2, h3, h4, h5, h6, p, div, ul, span": { color: "white" } })
