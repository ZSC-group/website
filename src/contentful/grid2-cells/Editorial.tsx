import { EditorialType } from "src/utils/contentful"
import { css } from "@emotion/react"
import { colors } from "src/colors"
import { flexRow, WHEN_DESKTOP, WHEN_MOBILE, WHEN_TABLET_AND_UP } from "src/estyles"
import { documentToReactComponents } from "@contentful/rich-text-react-renderer"
import Button, { SIZE } from "src/shared/Button.3"
import { standardStyles } from "src/styles"

type Props = EditorialType & { darkMode?: boolean }

export default function Editorial(props: Props) {
  return (
    <article css={rootCss}>
      <div css={css(innerCSS, props.darkMode && darkModeText)}>
        {documentToReactComponents(props.title)}
        <Button
          style={standardStyles.elementalMarginTop}
          href={props.button.fields.href || props.button.fields.assetLink?.fields?.file?.url}
          text={props.button.fields.words}
          kind={props.button.fields.kind}
          size={SIZE.normal}
          target={
            props.button.fields.assetLink?.fields?.file?.url ||
            (props.button.fields.href?.startsWith("http") && "_blank")
          }
        />
      </div>
      <img css={imageCss} src={props.image.fields.file.url} width={128} height={128} />
    </article>
  )
}

const rootCss = css(flexRow, {
  padding: "16px 8px 16px 24px",
  borderTop: `1px solid ${colors.gray}`,
  justifyContent: "space-between",
})

const innerCSS = css({
  marginRight: 16,
  flex: 1,
})

const imageCss = css({
  objectFit: "cover",
  width: 96,
  height: 96,
  [WHEN_DESKTOP]: {
    width: 128,
    height: 128,
  },
})

const darkModeText = css({ "h1, h2, h3, h4, h5, h6, p, div, ul, span": { color: "white" } })
