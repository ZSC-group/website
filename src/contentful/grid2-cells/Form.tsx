import { useForm } from "react-hook-form";
import {FormContentType} from "src/utils/contentful"
import { inputDarkStyle, labelStyle, WHEN_TABLET_AND_UP } from "src/estyles"
import { css } from "@emotion/react"
import Button, { BTN } from "src/shared/Button.3"
import * as React from "react"
import { postForm } from "src/forms/postForm"

export default function Form(props: FormContentType){

  const { register, handleSubmit, formState} = useForm();
  const onSubmit = data => postForm(props.route, data);

  const styles = React.useMemo(() => {
    return css(rootStyle, {
      gridColumn: `span ${props.colSpan}`,
      [WHEN_TABLET_AND_UP]: {
        gridTemplateAreas: props.layout?.grid?.map(row => row.join(" ")).map(e => `"${e}"`).join("\n")
      }
    }
  )},[props.colSpan, props.layout?.grid])

  return <form css={styles} onSubmit={handleSubmit(onSubmit)}>
      {props.fields.map(input => {
        const attributes = register(input.fields.name, {required:  input.fields.required})

        return <label key={input.sys.id} css={css(labelStyle, {gridArea:  props.layout ? input.fields.name : null})} >

          {input.fields.label}{input.fields.required && "*"}

          {input.fields.type === "multiline" ?
            <textarea  name={input.fields.name} css={inputDarkStyle} rows={5}
              {...attributes} />
            :
            <input name={input.fields.name} type={input.fields.type} css={inputDarkStyle}
              {...attributes}
            />
          }
        </label>
      }
      )}
      <span css={buttonCss}><Button  kind={BTN.PRIMARY} onPress={handleSubmit(onSubmit)} css={buttonCss} accessibilityRole="button" text={props.submitText}/></span>
  </form>
}

const buttonCss = css({
  gridArea: "button"
})

const rootStyle = css({
  [WHEN_TABLET_AND_UP]: {
    display: 'grid',
    marginTop: 24,
    columnGap: 24,
  }
})