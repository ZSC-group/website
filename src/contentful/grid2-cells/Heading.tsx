import { css } from "@emotion/react"
import { documentToReactComponents } from "@contentful/rich-text-react-renderer"
import { BLOCKS } from "@contentful/rich-text-types"

import { HeadingContentType } from "src/utils/contentful"
import { flex, fonts } from "src/estyles"

interface Props {
  span: number
  darkMode: boolean
}

export function Heading(props: Props & HeadingContentType) {
  function SubTitle(_, children) {
    return <h3 css={css(subtitleCss, props.subTitleCss)}>{children}</h3>
  }

  const subtitleOptions = {
    renderNode: {
      [BLOCKS.PARAGRAPH]: SubTitle,
    },
  }

  const imageFile = props.image?.fields?.file

  return (
    <div css={css(rootCss, { gridColumn: `span ${props.span}` })}>
      {imageFile && (
        <img
          src={imageFile.url}
          width={imageFile.details.image.width}
          height={imageFile.details.image.height}
        />
      )}
      {<h2 css={css(titleCSS, props.titleCss)}>{props.title}</h2>}
      {documentToReactComponents(props.subTitle, subtitleOptions)}
    </div>
  )
}

const rootCss = css(flex, {
  alignItems: "center",
  marginBottom: 80,
})

const subtitleCss = css(fonts.h4, {
  textAlign: "center",
})

const titleCSS = css(fonts.h1, {
  textAlign: "center",
  marginBottom: 24,
})
