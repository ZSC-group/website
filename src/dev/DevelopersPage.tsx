import { css } from "@emotion/react"
import * as React from "react"
import FullStack from "src/dev/FullStack"
import { flex } from "src/estyles"
import { Props } from "src/home/Home"
import { pageSwitch } from "src/public-sector/CommonContentFullPage"

export default function Developers(props: Props) {
  const items = props.sections.map(pageSwitch)
  items.splice(3, 0, <FullStack />)
  return (
    <>
      <div css={rootCss}>{items}</div>
    </>

  )
}

const rootCss = css(flex, {})


