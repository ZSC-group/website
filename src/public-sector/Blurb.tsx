import {css} from '@emotion/react'
import { Asset, Entry, } from 'contentful'
import * as React from 'react'
import {ContentfulButton} from "src/utils/contentful"
import {flex, fonts} from "src/estyles"
import Button, { SIZE } from 'src/shared/Button.3'
import { standardStyles } from 'src/styles'
import { documentToReactComponents } from '@contentful/rich-text-react-renderer'
import { Document } from '@contentful/rich-text-types'
import { BLOCKS } from '@contentful/rich-text-types'

enum Headings {
  "large" = 'large',
  "medium" = 'medium',
  "small" = 'small',
}

export interface Props {
  icon?: Asset
  title:string
  titleType?: Headings
  body: Document
  link?: Entry<ContentfulButton>
}

export default function Blurb(props: Props) {
  return <div css={rootCss}>
    <div css={containerCss}>
      <img src={props.icon?.fields?.file?.url} css={imageCss} width={100} height={100} />
      <h4 css={headingStyle(props.titleType)}>{props.title}</h4>
      {documentToReactComponents(props.body, {renderNode})}
    </div>
    {props.link && (
        <Button style={standardStyles.elementalMarginTop} href={props.link.fields.href} text={props.link.fields.words}  kind={props.link.fields.kind}  size={SIZE.normal}/>
      )}
  </div>
}

const rootCss = css (flex,{
  flex: 1,
  justifyContent: "space-between",
  marginBottom: 36
})

const containerCss = css(flex, {
  flex: 1
})

const imageCss =css({
  width: 100,
  height: 100
})

const headingCss = css({
  marginTop: 16,
  marginBottom: 12,
})

function headingStyle(type: Headings) {
  switch (type)  {
    case "large":
      return css(fonts.h4, headingCss)
    case "small":
      return css(fonts.h5, headingCss)
    default:
      return css(fonts.h3, headingCss)
  }
}


const renderNode = {
  [BLOCKS.PARAGRAPH]: (_, children: string) => {
    return <p css={bodyCss}>{children}</p>
  },
}
const bodyCss = css(fonts.body, {
  "&:first-of-type": {
    marginTop: 0
  },
  "&:last-of-type": {
    marginBottom: 0
  }
})