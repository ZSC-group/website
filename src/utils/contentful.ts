import { Document } from '@contentful/rich-text-types'
import { CSSObject } from '@emotion/react'
import { Asset, createClient, Entry, EntryCollection } from 'contentful'
import getConfig from 'next/config'
import { Page as SideBarEntry } from 'src/experience/common/Sidebar'
import  {Props as BlurbProps} from "src/public-sector/Blurb"
import { BTN } from 'src/shared/Button.3'

function intialize() {
  const { serverRuntimeConfig, publicRuntimeConfig } = getConfig()
  const isPreview = publicRuntimeConfig.ENV === 'development'
  return createClient({
    space: serverRuntimeConfig.CONTENTFUL_SPACE_ID,
    accessToken: isPreview
      ? serverRuntimeConfig.CONTENTFUL_PREVIEW_ACCESS_TOKEN
      : serverRuntimeConfig.CONTENTFUL_ACCESS_TOKEN,
    host: isPreview ? 'preview.contentful.com' : undefined,
  })
}

interface Kit {
  name: string
  slug: string
  metaDescription: string
  ogImage: Asset
  pages_: any[]
}

interface InternalKit {
  kitName: string
  metaDescription: string
  ogImage: Asset
  pageID: string
  sidebar: SideBarEntry[]
}

export async function getKit(kitSlug: string, pageSlug: string, { locale }): Promise<InternalKit> {
  const kit = await intialize().getEntries<Kit>({
    content_type: 'kit',
    'fields.slug': kitSlug,
    locale,
  })

  const data = kit.items[0].fields

  const homePageSlug = kitSlug === 'merchant' ? 'index' : kitSlug

  const actualPageSlug = !pageSlug ? homePageSlug : pageSlug

  const pageID = data.pages_.find((p) => p.fields.slug === actualPageSlug)?.sys?.id

  return {
    kitName: data.name,
    metaDescription: data.metaDescription,
    ogImage: data.ogImage,
    pageID,
    sidebar: data.pages_.map((page) => {
      return {
        title: page.fields.title,
        href: `/experience/${kitSlug}${
          page.fields.slug === kitSlug || page.fields.slug === 'index' ? '' : '/' + page.fields.slug
        }${addLocale(locale)}`,
        sections: [],
      }
    }),
  }
}


export interface SectionType { name: string; contentField: Document; slug: string }


export interface ContentfulButton {
  name: string
  href: string
  assetLink?: Asset
  words: string
  kind: BTN
}

export interface FreeContentType {
  backgroundColor: string
  colSpan?: 1 | 2 | 3 |4
  cssStyle: CSSObject
  body: Document
}

export type CellContentType = BlurbProps | FreeContentType

export interface GridRowContentType {
  id: string
  cells: Entry<CellContentType>[]
  cssStyle?: CSSObject
  columns: 1 | 2 | 3 | 4
  darkMode?: boolean
}



export interface ContentfulPage<T> {
  title: string
  slug: string
  description: string
  sections: Entry<T>[]
}

export async function getPageBySlug(slug: string, { locale }, showSysData?: boolean) {
  const pages = await intialize().getEntries<ContentfulPage<SectionType| GridRowContentType>>({
    content_type: 'page',
    'fields.slug': slug,
    include: 5,
    locale,
  })
  return processPages<SectionType| GridRowContentType>(pages, showSysData)
}

export async function getPageById<T>(id: string, { locale }) {
  const pages = await intialize().getEntries<ContentfulPage<T>>({
    content_type: 'page',
    'sys.id': id,
    include: 5,
    locale,
  })
  return processPages<T>(pages)
}

function processPages<T>(pages: EntryCollection<ContentfulPage<T>>, showSysData?: boolean) {
  const data = pages.items[0].fields
  const sections = showSysData ? data.sections : (data.sections || []).map((section) => section.fields)
  return { ...data, sections, updatedAt: pages.items[0].sys.updatedAt }
}

export function addLocale(locale) {
  if (locale === 'en-US') {
    return ''
  } else {
    return `?locale=${locale}`
  }
}

interface FAQItem {
  question: string
  answer: Document
}

interface FAQcollection {
  title: string
  list: Entry<FAQItem>[]
}

export async function getFAQ({ locale }) {
  const result = await intialize().getEntries<FAQcollection>({
    locale,
    content_type: 'faq',
    include: 3,
    'fields.key': 'celoFAQ',
  })
  const faqPage = result.items[0]
  return faqPage
}
