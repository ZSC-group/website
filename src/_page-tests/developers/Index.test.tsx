import * as React from "react"
import * as renderer from "react-test-renderer"
import { TestProvider } from "src/_page-tests/test-utils"
import DevelopersPage from "../../../pages/developers/index"
import { ContentfulPage, GridRowContentType } from "src/utils/contentful"
import { Props as BlurbProps } from "src/contentful/grid2-cells/Blurb"
import { Entry } from "contentful"
import { BLOCKS } from "@contentful/rich-text-types"

function blurbFactory(unique): Entry<BlurbProps> {
  return {
    sys: {
      id: unique,
      type: "",
      locale: "en",
      createdAt: "2021-07-01",
      updatedAt: "2021-07-01",
      contentType: { sys: { id: "proposition", type: "Link", linkType: "ContentType" } },
    },
    fields: {
      isNaturalSize: false,
      title: `Blurb ${unique}`,
      body: {
        nodeType: BLOCKS.DOCUMENT,
        content: [
          {
            nodeType: BLOCKS.PARAGRAPH,
            content: [{ nodeType: "text", value: "Hello World", marks: [], data: {} }],
            data: {},
          },
        ],
        data: {},
      },
    },
    toPlainObject: () => this,
    update: () => this,
  }
}

const TestData: ContentfulPage<GridRowContentType> = {
  title: "Celo: Build for all",
  description: "Celo's ultra light client makes it easy to build for smartphones across the world",
  slug: "developers",
  darkNav: false,
  sections: [
    {
      sys: {
        id: "1",
        type: "",
        locale: "en",
        createdAt: "2021-07-01",
        updatedAt: "2021-07-01",
        contentType: { sys: { id: "grid-row", type: "Link", linkType: "ContentType" } },
      },
      fields: {
        id: "test",
        columns: 3,
        cells: [blurbFactory("romeo"), blurbFactory("soma"), blurbFactory("rico")],
      },
      toPlainObject: () => this,
      update: () => this,
    },
  ],
}

describe("DevelopersPage", () => {
  it("renders", () => {
    const tree = renderer
      .create(
        <TestProvider>
          <DevelopersPage title={TestData.title} slug={TestData.slug} description={TestData.description} sections={TestData.sections} darkNav={TestData.darkNav} />
        </TestProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
