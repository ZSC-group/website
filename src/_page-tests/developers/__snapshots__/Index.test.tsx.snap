// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevelopersPage renders 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "1mju3xz",
      "next": undefined,
      "styles": "box-sizing:border-box;display:flex;flex-direction:column;;",
      "toString": [Function],
    }
  }
/>
`;
